commerce_coordinator/apps/frontend_app_payment/filters.py

```diff
@@ -81,3 +81,30 @@ def run_filter(cls, params):
         payment_output = DraftPaymentCreateViewOutputSerializer(data=payment)
         payment_output.is_valid(raise_exception=True)
         return payment_output.data
+
+
+class ActiveOrderRequested(OpenEdxPublicFilter):
+    """
+    Filter to gather payment data from the defined PipelineStep(s)
+    """
+
+    filter_type = "org.edx.coordinator.frontend_app_payment.active.order.requested.v1"
+
+    @classmethod
+    def run_filter(cls, params):
+        """
+        Call the PipelineStep(s) defined for this filter, to gather the current order
+        Arguments:
+            params: arguments passed through from the original get active order url querystring
+        """
+        active_order = super().run_pipeline(
+            edx_lms_user_id=params['edx_lms_user_id'],
+        )
+        # Transpose order data to the values frontend_app_payment expects
+        active_order['orderTotal'] = float(active_order.pop('total'))
```

Comment:
in our APIs, we are mostly using snake cases. so, `orderTotal` should be `order_total`. See [spec](https://github.com/edx/commerce-coordinator-private/blob/main/docs/api-spec/theseus-api-spec.md#output-13).
Also, can we use a serializer here? it will make it clean and the code will be re-useable.?

Reply:
Added an output serializer and changed the keys that were being added to snake case, however Titan is returning all keys in camel case are we expecting them to be in snake case or do you think we should parse through and change them to snake_case as they pass through coordinator?

Reply:
I think the best place to use GetActiveOrderOutputSerializer would be in GetTitanActiveOrder PipelineStep.
This serializer should be responsible to reformat keys to the snake case as well. The idea is to update the format as soon as we get a response from Titan.
This way we would be able to avoid manual data transposition in the following code block in the filter.

        # Transpose order data to the values frontend_app_payment expects
        active_order['order_total'] = float(active_order.pop('total'))
        # TODO: Do we still need this waffle flag? Normally from ecommerce
        # and was used from cybersource to stripe migration.
        active_order['enable_stripe_payment_processor'] = True
        active_order['products'] = active_order.pop('lineItems')
        active_order['basket_id'] = active_order.pop('uuid')
We have an example serializer [in titan app](https://github.com/edx/commerce-coordinator/blob/cb4dafb5e81c5dd7005d8075f220cb8570258982/commerce_coordinator/apps/titan/serializers.py#L20).